modules/runner/src/testFixtures/java/org/apache/ignite/internal/ClusterPerClassIntegrationTest.java

```diff
@@ -386,38 +384,37 @@ private static List<List<Object>> executeAwaitingIndexes(IgniteImpl node, Functi
         CatalogManager catalogManager = node.catalogManager();
 
         // Get existing indexes
-        Set<Integer> existing = catalogManager.indexes(catalogManager.latestCatalogVersion())
-                .stream().map(CatalogObjectDescriptor::id)
+        Set<Integer> existing = catalogManager.indexes(catalogManager.latestCatalogVersion()).stream()
+                .map(CatalogObjectDescriptor::id)
                 .collect(Collectors.toSet());
 
         List<List<Object>> result = statement.apply(node);
 
         // Get indexes after a statement and compute the difference
-        Set<Integer> difference = catalogManager.indexes(catalogManager.latestCatalogVersion()).stream()
+        List<Integer> difference = catalogManager.indexes(catalogManager.latestCatalogVersion()).stream()
                 .map(CatalogObjectDescriptor::id)
-                .collect(Collectors.toSet());
-
-        difference.removeAll(existing);
+                .filter(id -> !existing.contains(id))
+                .collect(Collectors.toList());
 
         if (difference.isEmpty()) {
             return result;
         }
 
         // If there are new indexes, wait for them to become available.
+        HybridClock clock = node.clock();
 
         try {
-            assertTrue(waitForCondition(() -> {
-                int latestVersion = catalogManager.latestCatalogVersion();
-                int notAvailable = 0;
-
-                for (CatalogIndexDescriptor index : catalogManager.indexes(latestVersion)) {
-                    if (!index.available() && difference.contains(index.id())) {
-                        notAvailable++;
-                    }
-                }
-
-                return notAvailable == 0;
-            }, 10_000));
+            assertTrue(waitForCondition(
+                    () -> {
+                        long now = clock.nowLong();
```

Comment:
Could you please add some comment to explain why we do it by such way

Reply:
Sure